Unknown Error in AddArrayReturnDocTypeRector

# Bug Report

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.9.2 / demo              |
| Installed as   | prefixed Rector / demo                           |

<!-- Please describe your problem here. -->
AddArrayReturnDocTypeRector errors with an unhelpful error
~~~json
{ "meta": { "version": "dev-master@8175188", "config": "/project/rector.php" }, "totals": { "changed_files": 0, "removed_and_added_files_count": 0, "removed_node_count": 0, "errors": 1 }, "errors": [ { "message": "", "file": "rector_analyzed_file.php", "line": 0 } ] }
~~~
or via cli
~~~
[ERROR] Could not process "filename.php" file, due to:
         "".
~~~

## Minimal PHP Code Causing Issue

https://getrector.org/demo/8ddce49d-3cd8-4f27-a99e-75fb6d8d58db

## Workaround

Manually add `/** @return array */` to docblock

--- a/packages/phpstan-static-type-mapper/src/TypeMapper/ClosureTypeMapper.php
+++ b/packages/phpstan-static-type-mapper/src/TypeMapper/ClosureTypeMapper.php
@@ -7,14 +7,22 @@ namespace Rector\PHPStanStaticTypeMapper\TypeMapper;
 use Closure;
 use PhpParser\Node;
 use PhpParser\Node\Name;
+use PHPStan\PhpDocParser\Ast\Type\IdentifierTypeNode;
 use PHPStan\PhpDocParser\Ast\Type\TypeNode;
 use PHPStan\Type\ClosureType;
 use PHPStan\Type\Type;
-use Rector\Core\Exception\NotImplementedException;
+use Rector\AttributeAwarePhpDoc\Ast\Type\AttributeAwareCallableTypeNode;
+use Rector\PHPStanStaticTypeMapper\Contract\PHPStanStaticTypeMapperAwareInterface;
 use Rector\PHPStanStaticTypeMapper\Contract\TypeMapperInterface;
+use Rector\PHPStanStaticTypeMapper\PHPStanStaticTypeMapper;
 
-final class ClosureTypeMapper implements TypeMapperInterface
+final class ClosureTypeMapper implements TypeMapperInterface, PHPStanStaticTypeMapperAwareInterface
 {
+    /**
+     * @var PHPStanStaticTypeMapper
+     */
+    private $phpStanStaticTypeMapper;
+
     public function getNodeClass(): string
     {
         return ClosureType::class;
@@ -25,7 +33,11 @@ final class ClosureTypeMapper implements TypeMapperInterface
      */
     public function mapToPHPStanPhpDocTypeNode(Type $type): TypeNode
     {
-        throw new NotImplementedException();
+        $identifierTypeNode = new IdentifierTypeNode($type->getClassName());
+
+        $returnDocTypeNode = $this->phpStanStaticTypeMapper->mapToPHPStanPhpDocTypeNode($type->getReturnType());
+
+        return new AttributeAwareCallableTypeNode($identifierTypeNode, [], $returnDocTypeNode);
     }
 
     /**
@@ -43,5 +55,10 @@ final class ClosureTypeMapper implements TypeMapperInterface
     public function mapToDocString(Type $type, ?Type $parentType = null): string
     {
         return '\\' . Closure::class;
+    }
+
+    public function setPHPStanStaticTypeMapper(PHPStanStaticTypeMapper $phpStanStaticTypeMapper): void
+    {
+        $this->phpStanStaticTypeMapper = $phpStanStaticTypeMapper;
     }
 }

--- a/rules/type-declaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/return_array_closure.php.inc
+++ b/rules/type-declaration/tests/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/return_array_closure.php.inc
@@ -1,0 +1,44 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+/**
+ * @var string $description
+ */
+class ReturnArrayClosure
+{
+    public function fields(): array
+    {
+        return [
+            'description' => function() {
+                return $this->description;
+            },
+        ];
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayReturnDocTypeRector\Fixture;
+
+/**
+ * @var string $description
+ */
+class ReturnArrayClosure
+{
+    /**
+     * @return array<string, Closure>
+     */
+    public function fields(): array
+    {
+        return [
+            'description' => function() {
+                return $this->description;
+            },
+        ];
+    }
+}
+
+?>

